aws-predictions/src/main/java/com/amplifyframework/predictions/aws/service/AWSComprehendService.java

```diff
@@ -0,0 +1,338 @@
+
+        // Detect sentiment from given text via AWS Comprehend
+        final DetectSentimentResult result;
+        try {
+            result = comprehend.detectSentiment(request);
+        } catch (AmazonClientException serviceException) {
+            throw new PredictionsException(
+                    "AWS Comprehend encountered an error while detecting sentiment.",
+                    serviceException,
+                    "See attached service exception for more details."
+            );
+        }
+
+        // Convert AWS Comprehend's detection result to Amplify-compatible format
+        String comprehendSentiment = result.getSentiment();
+        SentimentScore sentimentScore = result.getSentimentScore();
+        SentimentType predominantSentiment = SentimentTypeAdapter.fromComprehend(comprehendSentiment);
+        final float score;
+        switch (predominantSentiment) {
+            case POSITIVE:
+                score = sentimentScore.getPositive();
+                break;
+            case NEGATIVE:
+                score = sentimentScore.getNegative();
+                break;
+            case NEUTRAL:
+                score = sentimentScore.getNeutral();
+                break;
+            case MIXED:
+                score = sentimentScore.getMixed();
+                break;
+            default:
+                score = 0f;
+        }
+
+        return Sentiment.builder()
+                .value(predominantSentiment)
+                .confidence(score)
+                .build();
+    }
+
+    private List<KeyPhrase> fetchKeyPhrases(String text, LanguageType language) throws PredictionsException {
+        // Skip if configuration specifies NOT key phrase
+        if (!AWSInterpretConfiguration.InterpretType.ALL.equals(configuration.getType())
+                && !AWSInterpretConfiguration.InterpretType.KEY_PHRASE.equals(configuration.getType())) {
+            return null;
+        }
+
+        DetectKeyPhrasesRequest request = new DetectKeyPhrasesRequest()
+                .withText(text)
+                .withLanguageCode(language.getLanguageCode());
+
+        // Detect key phrases from given text via AWS Comprehend
+        final DetectKeyPhrasesResult result;
+        try {
+            result = comprehend.detectKeyPhrases(request);
+        } catch (AmazonClientException serviceException) {
+            throw new PredictionsException(
+                    "AWS Comprehend encountered an error while detecting key phrases.",
+                    serviceException,
+                    "See attached service exception for more details."
+            );
+        }
+
+        // Convert AWS Comprehend's detection result to Amplify-compatible format
+        List<KeyPhrase> keyPhrases = new ArrayList<>();
+        for (com.amazonaws.services.comprehend.model.KeyPhrase comprehendKeyPhrase : result.getKeyPhrases()) {
+            TargetText target = new TargetText(
+                    comprehendKeyPhrase.getText(),
```

Comment:
When using beginOffset and text, make sure that text containing special characters are counted correctly. For example text with smileys. "face \uD83D\uDE10" whats the length of this string? 17 or 7

Reply:
Applies to all places that uses beginOffset and length for text.

Reply:
Java appears to handle it appropriately actually (length 7). I will leave it as is for now.